flink-libraries/flink-python/src/main/java/org/apache/flink/python/api/PythonPlanBinder.java

```diff
@@ -94,6 +95,7 @@
 
 	private HashMap<Integer, Object> sets = new HashMap<>();
 	public ExecutionEnvironment env;
+	private int environmentCounter = 0;
```

Comment:
we should explicitly send the ID from python as a parameter like the parallelism. Right now would have to make sure that both python and java generate ID's the same way.